### ISSUE TITLE
Reverse Proxy

### ISSUE THREAD

**Issue by awptechnologies** [linkwarden/linkwarden#269]
How do i set this up to work with reverse proxy? i can log in with reverse proxy but when i log out it redirects me to localhost:3000 therefore never actually logging me out.

**Comment by awptechnologies** [comment]
i am using docker stack so .env files do not work so i had to list the individual environment variables to database and the linkwarden container. 

For postgress the env i have is 
POSTGRES_PASSWORD

For Linkwarden
DATABASE_URL
NEXTAUTH_URL
NEXTAUTH_SECRET
NEXT_PUBLIC_DISABLE_REGISTRATION

So far from what i can tell everything works correctly. The only issue i have is when i logout. Instead of redirecting me back to the login page i get an error with localhost:3000 in address bar. When i go back to the url given by reverse proxy it loads the login page perfect.

**Comment by cardi-m** [comment]
Have u tried to set

''NEXTAUTH_URL=http://localhost:3000''

instead of Localhost the Doumain you set in the reverse Proxy?

**Comment by awptechnologies** [comment]
yes that is what i use.... that's the only way the reverse proxy will work. My reverse proxy uses https://mydomain.domain.com I use haproxy so the proxy is pointed at host-ip:3000. Loading page and login in works fine, i'm able to save links and everything seems to be working right. The issue is when i log out. normally when you log it of a website you are redirected back to the login page. But somehow it pulls the localhost:3000 from config and redirects me to there. Since the machine i'm running the browser on is not the host that hosts linkwarden it gives me an error. I don't believe it is my reverse proxy causing the problem i run 20 other services all using same configs ranging from portainer to uptime kuma.  and with all of them when i logout i am redirected to login page using the actuall domain.

**Comment by awptechnologies** [comment]
also the browser extension is working fine. i just think there is something coded in the container that redirects you to localhost:3000 on logout.

**Comment by falkheiland** [comment]
> How do i set this up to work with reverse proxy? i can log in with reverse proxy but when i log out it redirects me to localhost:3000 therefore never actually logging me out.

same here - using traefik

**Comment by vincent1890** [comment]
same here - using Nginx Proxy

`
The page is not redirected correctly

Firefox has detected that the server redirects the request for this address in a way that will not succeed.

    The cause of this problem may be the deactivation or refusal of cookies.
`

**Comment by daniel31x13** [comment]
This is probably because you should set `NEXTAUTH_URL_INTERNAL` as well: 
https://next-auth.js.org/configuration/options#nextauth_url_internal

So your environment variables should look something like:

```
NEXTAUTH_URL=https://example.com
NEXTAUTH_URL_INTERNAL=http://localhost:3000

rest of .env file...
```

Just make sure to set the port as the internal port used in docker...

**Comment by tarocjsu** [comment]
Nginx Proxy Manager or Traefik can be work for reverse proxy.

Your guys need to google or study it by yourself first, this is not issue or request.

**Comment by RedWavve** [comment]
+1 on this issue.
I have 5 proxies in NGINX, and Linkwarden is the only one that refuses to work behind a proxy when the port 3000 is forwarded towards NGINX.
It works perfectly when not going through a proxy.

NGINX v2.11.2
Linkwarden v2.5.3 via Unraid-Docker

**Comment by Snuupy** [comment]
for swag:

```
server {
    listen 443 ssl;
    listen [::]:443 ssl;

    server_name linkwarden.*;

    include /config/nginx/ssl.conf;

    # client_max_body_size 0;

    location / {
        include /config/nginx/proxy.conf;
        include /config/nginx/resolver.conf;
        set $upstream_app linkwarden;
        set $upstream_port 3000;
        set $upstream_proto http;
        proxy_set_header Host $host; 
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for; 
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection $connection_upgrade;
        proxy_pass $upstream_proto://$upstream_app:$upstream_port;
        }
}

```

**Comment by fabienpascal** [comment]
The self-hosted application doesn't use linkwarden as the docker container name. It uses set linkwarden-linkwarden-1.
So make sure to use this as the app name for $upstream_app
